actor/builtin/storagemarket/storagemarket.go

```diff
@@ -237,6 +288,19 @@ func (sma *Actor) GetProofsMode(vmctx exec.VMContext) (types.ProofsMode, uint8,
 	return size, 0, nil
 }
 
+func (sma *Actor) getMinerIsProving(vmctx exec.VMContext, minerAddr address.Address) (bool, error) {
+	msgResult, _, err := vmctx.Send(minerAddr, "isProving", types.ZeroAttoFIL, nil)
+	if err != nil {
+		return false, err
+	}
+
+	isProving, err := abi.Deserialize(msgResult[0], abi.Boolean)
+	if err != nil {
+		return false, err
+	}
+	return isProving.Val.(bool), nil
+}
```

Comment:
@acruikshank FWIW I think it's a bit silly that read-only methods that are not part of consensus have to go through this message-passing and ABI-deserialization junk. We're doing something wrong.

Reply:
(non-blocking)
I agree, but it's also kind of silly that state-manipulating methods that are part of consensus go through ABI-deserialization. Nothing about consensus requires this right now. It's all part of a long term plan to move the state machine into a separate VM that we haven't been able to prioritize.

There are advantages to keeping code that accesses actor-state within the actor. There are also advantages to limiting actors to message processing (actors not objects). There's no real advantage to serializing parameters and deserializing results (until we have a VM). We can simplify this interface in the near term.

One option would be a function that could generate a populated VMContext at the chain head. With this, you could call the actor method directly. See #3067 for an example of doing this in tests.